code/modules/GOAP/cyka_blyat/actions.dm

```diff
@@ -0,0 +1,339 @@
+		var/obj/item/russian_reload/RE = new /obj/item/russian_reload(get_turf(AA_MATCH_TARGET_OVERLAYS))
+		RE.throw_at(target, 10, 1)
+		RU.reloads_left = 3
+		RU.say(pick(list("Spasiba!","Spasiba za boyepripasy!","Ti khahroshiy spetsiahlist!")))
+		QDEL_IN(RE, 15)
+		action_done = TRUE
+		return ..()
+	else
+		action_done = TRUE
+		return FALSE
+
+
+/datum/goap_action/russian/resupply/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/russian/dodge
+	name = "Dodge!"
+	cost = 1
+	cooldown = 50
+	saytext = "OPAH! CANNOT EVEN HIT RUSSIAN BABY!"
+
+/datum/goap_action/russian/dodge/New()
+	..()
+	preconditions = list()
+	preconditions["dodgeEnemy"] = FALSE
+	effects = list()
+	effects["dodgeEnemy"] = TRUE
+
+/datum/goap_action/russian/dodge/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/carbon/C in viewl)
+		if(C && !C.stat)
+			target = C
+	return (target != null)
+
+/datum/goap_action/russian/dodge/RequiresInRange(atom/agent)
+	return FALSE
+
+/datum/goap_action/russian/dodge/Perform(mob/living/simple_animal/hostile/H)
+	H.SpinAnimation(5, 0) // half a second spin but one second immunity
+	DodgeEffect(H)
+	var/oldhealth = H.health
+	H.health = INFINITY // Iframes!
+	addtimer(VARSET_CALLBACK(H, health, oldhealth), 10) // turn their health back
+	action_done = TRUE
+	return ..() 
+
+/datum/goap_action/russian/dodge/proc/DodgeEffect(atom/agent)
+	var/obj/effect/temp_visual/decoy/fading/halfsecond/N = new (get_turf(agent), agent)
+	var/obj/effect/temp_visual/decoy/fading/halfsecond/E = new (get_turf(agent), agent)
+	var/obj/effect/temp_visual/decoy/fading/halfsecond/S = new (get_turf(agent), agent)
+	var/obj/effect/temp_visual/decoy/fading/halfsecond/W = new (get_turf(agent), agent)
+	step(N, NORTH)
+	step(E, EAST)
+	step(S, SOUTH)
+	step(W, WEST)
+
+/datum/goap_action/russian/dodge/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/russian/throw_knives
+	name = "Throw a knife"
+	cost = 1// attack + dodge = 6 so this is only done if you can't attack
+	cooldown = 100
+	saytext = "I SHARE KNIFE WITH YOU!"
+
+/datum/goap_action/russian/throw_knives/New()
+	..()
+	preconditions = list()
+	preconditions["enemyDead"] = FALSE
+	effects = list()
+	effects["enemyDead"] = TRUE
+
+/datum/goap_action/russian/throw_knives/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/carbon/C in viewl)
+		if(C && !C.stat)
+			target = C
+	return (target != null)
+
+/datum/goap_action/russian/throw_knives/RequiresInRange(atom/agent)
+	return FALSE
+
+/datum/goap_action/russian/throw_knives/Perform(atom/agent)
+	var/obj/item/kitchen/knife/K = new /obj/item/kitchen/knife(get_turf(agent))
+	K.throw_at(target, 10, 1, agent)
+	action_done = TRUE
+	return ..() 
+
+/datum/goap_action/russian/throw_knives/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/russian/melee
+	name = "Melee"
+	cost = 2
+	saytext = "Bleed, american pig!"
```

Comment:
*capitalist, America :tm: doesn't exist

Reply:
What about Space America, land of the space free?